### ISSUE TITLE
Build dependencies cannot be hidden

### ISSUE THREAD

**Issue by ocaisa** [hpcugent/easybuild-framework#1368]
(Transferred from  hpcugent/easybuild-easyconfigs#1883)

The following in an easyconfig will not work:
```
# zlib required, but being linked instatically, so not a runtime dep
builddependencies = [('zlib', '1.2.8')]
hiddendependencies = [('zlib', '1.2.8')]
```
This results in:
```
-bash-4.2$ eb --force --hidden ./binutils-2.25.eb 
== temporary log file in case of crash /tmp/eb-KYLZRy/easybuild-Ydxdg_.log
ERROR: Failed to process easyconfig /homeb/zam/swmanage/BINUTILS/binutils-2.25.eb: Hidden dependencies with visible module names ['Core/zlib/1.2.8'] not in list of dependencies: []
```

**Comment by boegel** [comment]
I just realised that this would be very useful to hide the modules that are needed to bootstrap the GCC+binutils combo in the GNU toolchain; it's one of the few locations where marking dependencies as hidden by default makes sense

**Comment by boegel** [comment]
Not trivial due to this (copy-pasted from https://github.com/hpcugent/easybuild-easyconfigs/issues/1883):

Hmm, I guess we should simply relax the requirement and also accept that hidden dependencies are included in `builddependencies`. This does complicate dumping the easyconfig though, since `hiddendependencies` need to be restored in either `dependencies` or `builddependencies` then. See #1354.

Maybe we can simply introduce `hiddenbuilddependencies`, but I don't like where this is going...

**Comment by JackPerdue** [comment]
As I sit here pondering my own variation on the above mentioned bootstrap (which now gets up to binutilsbase/2.25-gccbase-4.8.4-binutilsbase-2.25-gccbase-4.4.7-11.el6 before I build a 'GCC' [be grateful I didn't include the RHEL6 binutils version in there too]), we really need this.  Hiddendeps are great but so long as they are forced into deps and not simply builddeps, it makes it hard to make the last jump where I can hide the bases from the user.

/me casts two votes (two hands up) for '''hiddenbuilddependencies'''.

**Comment by JackPerdue** [comment]
Note that this especially makes sense on the BG/Q where everything is usually compiled statically anyway (no runtime deps).

**Comment by boegel** [comment]
a fix for this is in #1503, let's follow up there (I want to avoid having duplicate issues open)
